esx_service/vmdk_ops.py

```diff
@@ -600,14 +598,16 @@ def removeVMDK(vmdk_path, vol_name=None, vm_name=None, tenant_uuid=None, datasto
 
     # Check the current volume status
     kv_status_attached, kv_uuid, attach_mode, attached_vm_name = getStatusAttached(vmdk_path)
+    # If the volume is attached then don't try removing the volume, the uuid may be
+    # resolved to a VM on the local host or may not if the VM is on another host,
+    # either case attempting to clean the VMDK can leave it in an inconsistent state,
+    # VMDK is present but the KV has been deleted.
     if kv_status_attached:
-        ret = handle_stale_attach(vmdk_path, kv_uuid)
-        if ret:
-            if vol_name is None:
-                vol_name = vmdk_utils.get_volname_from_vmdk_path(vmdk_path)
-            logging.info("*** removeVMDK: %s is in use, volume = %s VM = %s VM-uuid = %s (%s)",
-                vmdk_path, vol_name, attached_vm_name, kv_uuid, ret)
-            return err("Failed to remove volume {0}, in use by VM = {1}.".format(vol_name, attached_vm_name))
+        if vol_name is None:
+            vol_name = vmdk_utils.get_volname_from_vmdk_path(vmdk_path)
+        logging.info("*** removeVMDK: %s is in use, volume = %s VM = %s VM-uuid = %s",
+            vmdk_path, vol_name, attached_vm_name, kv_uuid)
+        return err("Failed to remove volume {0}, in use by VM = {1}.".format(vol_name, attached_vm_name))
```

Comment:
AFAICT we are not relying on data in KV for any operation. This would be an exception.

If I understand correctly this is a workaround for DiskLib issue. Rather than adding a workaround here, I would propose to follow solution proposed in original bug. Read KV in memory, delete file, if delete fails, write KV back to disk.